pkg/security/x509source.go

```diff
@@ -277,6 +316,91 @@ func (x *x509source) requestFromSentry(ctx context.Context, csrDER []byte) ([]*x
 	return workloadcert, nil
 }
 
+// maybeWriteSVIDToDir writes the current SVID to the configured directory if
+// configured. Does this atomically using unix symlinks.
+func (x *x509source) maybeWriteSVIDToDir() error {
+	if x.writeToDiskDir == nil || x.currentSVID == nil || x.trustAnchors == nil {
+		return nil
+	}
+
+	certs, pk, err := x.currentSVID.Marshal()
+	if err != nil {
+		return err
+	}
+
+	ta, err := x.trustAnchors.Marshal()
+	if err != nil {
+		return err
+	}
+
+	if err := atomicWrite(x.clock, *x.writeToDiskDir, map[string][]byte{
+		"tls.crt": certs,
+		"tls.key": pk,
+		"ca.crt":  ta,
+	}); err != nil {
+		return fmt.Errorf("unable to write SVID to disk: %w", err)
+	}
+
+	log.Infof("Wrote SVID to %q", *x.writeToDiskDir)
+
+	return nil
+}
+
+func (x *x509source) updateTrustAnchorFromFile(filepath string) error {
+	x.lock.RLock()
+	defer x.lock.RUnlock()
+
+	rootPEMs, err := os.ReadFile(filepath)
+	if err != nil {
+		return fmt.Errorf("failed to read trust anchors file %q: %s", filepath, err)
+	}
+
+	trustAnchorCerts, err := secpem.DecodePEMCertificates(rootPEMs)
+	if err != nil {
+		return fmt.Errorf("failed to decode trust anchors: %s", err)
+	}
+
+	x.trustAnchors.SetX509Authorities(trustAnchorCerts)
+
+	if err := x.maybeWriteSVIDToDir(); err != nil {
+		return err
+	}
+
+	for _, ch := range x.trustAnchorSubscribers {
+		go func(chi chan<- struct{}) { chi <- struct{}{} }(ch)
+	}
+
+	return nil
+}
+
+func atomicWrite(clock clock.Clock, dir string, data map[string][]byte) error {
+	dir = filepath.Clean(dir)
+	if err := os.MkdirAll(filepath.Dir(dir), 0o700); err != nil {
+		return err
+	}
+
+	newDir := dir + "-" + clock.Now().Format(time.RFC3339)
```

Comment:
maybe not too important in the container, but should we remove the dir on error?